ossfuzz/compress_fuzzer.cc

```diff
@@ -0,0 +1,22 @@
+#include <stddef.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include "lz4.h"
+
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size)
```

Comment:
This project is otherwise entirely in `C`, and this file doesn't use any C++. Can you please just make this a `.c` file, and remove `extern "C"`.

Making the fuzzers `C` also makes fuzzing with MSAN much, much easier, since we don't have to worry about compiling the C++ standard library in MSAN mode.

Reply:
So there is method to my madness, in that when I've tried this in pure C before (in curl) it just plain didn't work. Happy to explore going pure C though!!

Reply:
Yep, trying again with just `clang`, we start running into fun.
```
# Now compile the actual fuzzer.
clang -O1 -fno-omit-frame-pointer -gline-tables-only -DFUZZING_BUILD_MODE_UNSAFE_FOR_PRODUCTION -fsanitize=address -fsanitize-address-use-after-scope -fsanitize=fuzzer-no-link -I../lib -g -DLZ4_DEBUG=1  -DXXH_NAMESPACE=LZ4_  -fsanitize=fuzzer compress_fuzzer.o ../lib/liblz4.a -o compress_fuzzer
/usr/bin/ld: cannot find -lstdc++
```
However, installing libstdc++-5-dev seems to get rid of those errors. So I think I'm happy to change the scripts to do that. Worth noting that I haven't explicitly tested all the other options for building images.

Reply:
Actually, retesting with AFL, it doesn't obviously compile with clang. Will test if C++ actually improves that situation.

Reply:
Yep, with C++ it works fine. So if you want more fuzzer coverage I'd stick with the C++ version

Reply:
The zstd OSS-fuzz fuzzers are written in C, so I know it can work.

Reply:
Yep, as discussed below, all fine if you just link with C++ rather than compile.